### DISCUSSION TITLE
Spot time gap in TMY files for pre-processor prepare meteorological data

### DISCUSSION THREAD

**Discussion by nikkopante** [UMEP-dev/UMEP#483] [Q&A] [answered]
**Describe the bug**
Hi UMEP team, I'd like to ask question on how to spot time gap in the pre-processor prepare meteorological data. I download a TMY from the [NSRDB](https://nsrdb.nrel.gov/data-viewer). I want to convert it to UMEP acceptable `.txt` format. I am confronted with this error below. However, I cannot spot what is missing with the time gap. Here is my [data](https://drive.google.com/file/d/1SyrvFq7XzzqJUM2ZiDRCVkQVI-UWtUxT/view?usp=share_link). I'd like to ask if there's a minimum number of entries in the TMY.

![image](https://user-images.githubusercontent.com/35827302/205318179-abe2ab10-47e6-47f1-911d-47cffe48abad.png)

**To Reproduce**
1. UMEP > Pre-processor > Meteorological Data > Prepare Existing Data
2. Supply the needed column i.e. Year, Month, etc.
3. Export data
4. Error

**Desktop (please complete the following information):**
 - OS: Windows 10
 - QGIS Version: 3.22.10-Białowieża
 - UMEP Version: 3.20.9

**Comment by biglimp** [comment, marked as answer]
The reason is that your year column (first column) include months from multiple years. This is not allowed in UMEP as some tools require continuous data. Change the fist column to only include e.g. 2014 (except fro the last line that should be 2015), then it should work.

**Reply by nikkopante** [reply]
Thank you @biglimp !
